service/options_test.go

```diff
@@ -32,27 +32,26 @@ import (
 	"github.com/uber-go/tally"
 )
 
-func TestAddModules_OK(t *testing.T) {
-	sh := &host{}
-	require.NoError(t, sh.AddModules(successModuleCreate))
-	assert.Empty(t, sh.Modules())
+func TestNewOwner_ModulesOK(t *testing.T) {
+	_, err := newOwner([]ModuleCreateFunc{successModuleCreate}, withConfig(validServiceConfig))
+	require.NoError(t, err)
 }
 
-func TestAddModules_Errors(t *testing.T) {
-	sh := &host{}
-	assert.Error(t, sh.AddModules(errorModuleCreate))
+func TestNewOwner_ModulesErr(t *testing.T) {
+	_, err := newOwner([]ModuleCreateFunc{errorModuleCreate}, withConfig(validServiceConfig))
+	assert.Error(t, err)
 }
 
-func TestWithMetrics_OK(t *testing.T) {
+func TestNewOwner_WithMetricsOK(t *testing.T) {
 	assert.NotPanics(t, func() {
-		New(WithMetrics(tally.NoopScope, metrics.NopCachedStatsReporter))
+		newOwner([]ModuleCreateFunc{successModuleCreate}, WithMetrics(tally.NoopScope, metrics.NopCachedStatsReporter))
 	})
 }
 
-func TestWithTracing_OK(t *testing.T) {
+func TestNewOwner_WithTracingOK(t *testing.T) {
 	tracer := &opentracing.NoopTracer{}
 	assert.NotPanics(t, func() {
-		New(WithTracer(tracer))
+		newOwner([]ModuleCreateFunc{successModuleCreate}, WithTracer(tracer))
```

Comment:
can we pass `nil` instead of the success? these tests just got a lot harder to read. wondering if we can pass an empty slice of module create funcs, or whether the framework disallows that